prism/src/main/java/org/apache/falcon/resource/AbstractInstanceManager.java

```diff
@@ -173,9 +174,16 @@ public InstancesResult getStatus(String type, String entity, String startStr, St
 
             // LifeCycle lifeCycleObject = EntityUtil.getLifeCycle(lifeCycle);
             AbstractWorkflowEngine wfEngine = getWorkflowEngine(entityObject);
-            return getInstanceResultSubset(wfEngine.getStatus(entityObject,
-                            startAndEndDate.first, startAndEndDate.second, lifeCycles, allAttempts),
+            InstancesResult instancesResult = getInstanceResultSubset(wfEngine.getStatus(entityObject,
```

Comment:
Doesn't getInstanceResultSubset already apply numResults before returning?  If there are i1-i20 instances and if startTime is i11 and numResults is 5, does this code return: i16 - i20  OR i11 - i15 ?

From the code changes, I think it will be i16-i20. But, what Pragya is asking for is i11-i15. Can you please confirm?